worker/task_executor.go

```diff
@@ -0,0 +1,164 @@
+package worker
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	TaskScriptExecutorFileName  = "task-script-executor.sh"
+	RunTaskScriptCommandPattern = "/bin/bash %s -d -tsf=%s"
+	DefaultWorkerDockerImage    = "ubuntu"
+)
+
+type TaskExecutor struct {
+	Cli client.Client
+	Cid string
+}
+
+func (e *TaskExecutor) Execute(task *Task, containerSpawnWarner chan<- interface{}) error {
+	image := task.Image
+
+	if image == "" {
+		image = DefaultWorkerDockerImage
+	}
+	log.Println("Creating container with image: " + image)
+
+	config := utils.ContainerConfig{
+		Name:   "",
```

Comment:
what happens when we use an empty name? I'm wondering if we could create a name based on taskId and a random string appended to the taskId (this could ease the debug)

Reply:
consider it done.